arch/arm/core/cortex_m/Kconfig

```diff
@@ -203,7 +203,7 @@ config SW_VECTOR_RELAY
 	prompt "Enable Cortex-M0 Vector Table soft relay"
 	default n
 	default y if BOOTLOADER_MCUBOOT
-	depends on CPU_CORTEX_M0 && !CPU_CORTEX_M0_HAS_VECTOR_TABLE_REMAP
+	depends on CPU_CORTEX_M0 && !(CPU_CORTEX_M_HAS_VTOR || CPU_CORTEX_M0_HAS_VECTOR_TABLE_REMAP)
```

Comment:
@carlescufi please clarify: can the M0 have a VTOR register? Or is it part of M0+, and ARMv7?

Reply:
The M0+ has a VTOR register, the M0 does not. Since we do not have an M0_PLUS Kconfig we use this.

Reply:
ARMv7-M always has VTOR

Reply:
LGTM

Reply:
The M0 cannot have VTOR, the M0+ _may_ have VTOR.